src/uu/rm/src/rm.rs

```diff
@@ -399,14 +408,54 @@ fn remove_dir(path: &Path, options: &Options) -> bool {
                     show_error!("cannot remove {}: Directory not empty", path.quote());
                     return true;
                 }
+            }
+            Err(e) => {
+                if e.kind() == std::io::ErrorKind::PermissionDenied && options.dir {
+                    // try to see if dir to be removed is empty
+                    return remove_empty_dir_no_perm(path, options);
+                } else {
+                    show_error!("cannot remove {}: {}", path.quote(), e);
+                    return true;
+                }
+            }
+        }
+    }
+
+    false
+}
+
+fn remove_empty_dir_no_perm(path: &Path, options: &Options) -> bool {
+    match fs::metadata(path) {
+        Ok(metadata) => {
+            let original_perm = metadata.permissions();
+            #[cfg(unix)]
+            fs::set_permissions(path, fs::Permissions::from_mode(0o777)).unwrap();
```

Comment:
you aren't managing the potential error here, is it expected ?
thanks

Reply:
I should prob handle that